common/crc.c

```diff
@@ -54,3 +92,12 @@ uint32_t CRC8Maxim(uint8_t *buff, size_t size)
 	}
 	return crc_finish(&crc);
 }
+
+// width=8 poly=0x1d, init=0xc7 (0xe3 - WRONG! but it mentioned in MAD datasheet) refin=false  refout=false  xorout=0x00 name="CRC-8/MIFARE-MAD"
```

Comment:
0xc7 = 11000111
0xe3 = 11100011 = reflect(0xc7, 8)
You should be able to use the existing `crc_update()` (shifts right) instead of `crc_update2()` (shifts left), when you reflect polynom and start value and finally reflect the CRC.

Reply:
ill try it on monday. dont have valid mad at home(